servicesExternal/ipfs_service.py

```diff
@@ -1,21 +1,43 @@
 import ipfsapi
+import json
 
 from config import config
 
 # Note to contributers: to work on this you will need to use
 # Python's asyncio module most likely
 
 class Ipfs:
-    def __init__(self, publicIpfs: bool=True):
-        pass
+
+    _instance = None
+
+    @staticmethod
+    def init(publicIpfs: bool=True):
+        Ipfs._instance = Ipfs(publicIpfs)
 
     @staticmethod
-    def getInstance():
-        pass
+    def get_instance():
+        return Ipfs._instance
+
+    def __init__(self, publicIpfs: bool=True):
+        self.ipfsConfig = config["ipfs"]["nodeUrlDefault"]["public" if publicIpfs else "private"]
+        self.ipfs = ipfsapi.connect(self.ipfsConfig["host"], self.ipfsConfig["port"], self.ipfsConfig["protocol"])
+
+    async def add_file(self, data: str):
+        if data == None or data == "":
+            raise ValueError("")
+        
+        data_parsed = json.loads(data)
+        result = await self.ipfs.add(json.dumps(data_parsed, separators=(',',':')))
 
-    def addFile(self, data: str):
-        pass
+        return result["hash"]
 
-    def getFile(self, hash: str):
-        pass
+    async def get_file(self, hash: str):
+        if hash == None or hash == "":
+            raise ValueError("")
+        
+        try:
+            result = await self.ipfs.cat(hash)
 
+            return result
+        except ipfsapi.exceptions.Error as err:
+            raise err
```

Comment:
Again maybe try to give some more details about what might have happened so people can debug